pkg/authorizationserver/oauth2.go

```diff
@@ -1,15 +1,52 @@
 package authorizationserver
 
 import (
+	"crypto/rand"
+	"crypto/rsa"
 	"net/http"
 	"os"
+	"time"
 
+	"github.com/TBD54566975/ssi-sdk/oidc/issuance"
 	"github.com/ardanlabs/conf"
+	"github.com/goccy/go-json"
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/compose"
+	"github.com/ory/fosite/handler/openid"
+	"github.com/ory/fosite/storage"
+	"github.com/ory/fosite/token/jwt"
 	"github.com/tbd54566975/ssi-service/config"
 	"github.com/tbd54566975/ssi-service/pkg/server/framework"
 	"github.com/tbd54566975/ssi-service/pkg/server/middleware"
 )
 
+// A session is passed from the `/auth` to the `/token` endpoint. You probably want to store data like: "Who made the request",
+// "What organization does that person belong to" and so on.
+// For our use case, the session will meet the requirements imposed by JWT access tokens, HMAC access tokens and OpenID Connect
+// ID Tokens plus a custom field
+
+// newSession is a helper function for creating a new session. This may look like a lot of code but since we are
+// setting up multiple strategies it is a bit longer.
+// Usually, you could do:
+//
+//	session = new(fosite.DefaultSession)
+func newSession(user string) *openid.DefaultSession {
+	return &openid.DefaultSession{
+		Claims: &jwt.IDTokenClaims{
+			Issuer:      "https://fosite.my-application.com",
+			Subject:     user,
+			Audience:    []string{"https://my-client.my-application.com"},
+			ExpiresAt:   time.Now().Add(time.Hour * 6),
```

Comment:
may want a default  or configurable expiry time

Reply:
Same as above, will evolve in a future PR.